Tests/Integration/Core/VaultTest.php

```diff
@@ -0,0 +1,210 @@
+    {
+        $oUser = $this->getMockBuilder(User::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getId', 'getSelectedAddressId'])
+            ->getMock();
+        $oUser->method('getId')
+            ->willReturn($sUserId);
+        $oUser->method('getSelectedAddressId')
+            ->willReturn($sAddressId);
+
+        $this->getSession()->setUser($oUser);
+
+        $aCards = Vault::getCards();
+        $this->assertEquals($aExpected, $aCards);
+    }
+
+    public function getCardsProvider()
+    {
+        $oFutureDate = new DateTime();
+        $oFutureDate->add(new DateInterval('P1Y'));
+        $iYear = (int) $oFutureDate->format('y');
+        return [
+            'User 1 only one valid card' => [
+                'User ID 1',
+                'Address ID 1',
+                [
+                    [
+                        'OXID' => 1,
+                        'USERID' => 'User ID 1',
+                        'ADDRESSID' => 'Address ID 1',
+                        'TOKEN' => 'Token 1',
+                        'MASKEDPAN' => 'Masked****Pan',
+                        'EXPIRATIONMONTH' => 2,
+                        'EXPIRATIONYEAR' => $iYear,
+                    ],
+                ],
+            ],
+            'User 2 with 2 valid cards' => [
+                'User ID 2',
+                'Address ID 2',
+                [
+                    [
+                        'OXID' => 3,
+                        'USERID' => 'User ID 2',
+                        'ADDRESSID' => 'Address ID 2',
+                        'TOKEN' => 'Token 3',
+                        'MASKEDPAN' => 'Masked****Pan',
+                        'EXPIRATIONMONTH' => 2,
+                        'EXPIRATIONYEAR' => $iYear,
+                    ],
+                    [
+                        'OXID' => 4,
+                        'USERID' => 'User ID 2',
+                        'ADDRESSID' => 'Address ID 2',
+                        'TOKEN' => 'Token 4',
+                        'MASKEDPAN' => 'Masked****Pan',
+                        'EXPIRATIONMONTH' => 3,
+                        'EXPIRATIONYEAR' => $iYear,
+                    ],
+                ],
+            ],
+            'User 3 without valid card' => [
+                'User ID 3',
+                'Address ID 3',
+                [],
+            ],
+            'User 5 without saved card' => [
+                'User ID 5',
+                'Address ID 5',
+                [],
+            ],
+        ];
+    }
+
+    public function testDeleteCard()
+    {
+        $oUser = $this->getMockBuilder(\OxidEsales\EshopCommunity\Application\Model\User::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $oUser->method('getId')
+            ->willReturn('User ID 2');
+
+        $this->getSession()->setUser($oUser);
+        Vault::deleteCard(4);
+        $aResult = $this->getDb(DatabaseInterface::FETCH_MODE_ASSOC)->getAll(
```

Comment:
create a private method for getting the cards

Reply:
👍